.ci/go-test.sh

```diff
@@ -249,13 +246,21 @@ main()
 
 	[ -z "$test_packages" ] && echo "INFO: no golang code to test" && exit 0
 
+	# directory to use for temporary files
+	golang_tmp=$(mktemp -d)
+
+	# KATA_GO_TEST_FLAGS can be set to change the flags passed to "go test".
+	go_test_flags=${KATA_GO_TEST_FLAGS:-"-v $race -timeout $timeout_value -outputdir \"${golang_tmp}\""}
+
 	if [ "$1" = "html-coverage" ]; then
 		test_html_coverage
 	elif [ "$run_coverage" = yes ]; then
 		test_coverage
 	else
 		test_local
 	fi
+
+	[ -d "${golang_tmp}" ] && [ "${golang_tmp}" != "/" ] && rm -rf "${golang_tmp}"
```

Comment:
as the script has a `-e` set, would this be safer as a shell TRAP to catch the case where the tests fail?

Reply:
I did consider that but thought it would be better to leave the tmpfiles in the case of test failures as they might help us diagnose what went wrong.